libraries/standard/http/utest/http_utest.c

```diff
@@ -285,39 +280,38 @@ void test_Http_InitializeRequestHeaders_invalid_params()
  * header is set to "keep-alive" when HTTP_REQUEST_KEEP_ALIVE_FLAG in requestHeaders
  * is activated.
  */
-void test_Http_InitializeRequestHeaders_req_info()
+void test_Http_InitializeRequestHeaders_Req_info()
 {
-    HTTPStatus_t test_err = HTTP_INTERNAL_ERROR;
+    HTTPStatus_t httpStatus = HTTP_INTERNAL_ERROR;
     HTTPRequestHeaders_t requestHeaders = { 0 };
     HTTPRequestInfo_t requestInfo = { 0 };
-    size_t expectedHeaderLen = HTTP_TEST_MAX_INITIALIZED_HEADER_LEN;
-    char expectedHeader[ HTTP_TEST_INITIALIZED_HEADER_BUFFER_LEN ] = { 0 };
     int numBytes = 0;
 
     setupRequestInfo( &requestInfo );
-    setupBuffer( &requestHeaders, expectedHeaderLen );
+    expectedHeaders.dataLen = HTTP_TEST_PREFIX_HEADER_LEN -
+                              HTTP_TEST_REQUEST_PATH_LEN +
+                              HTTP_EMPTY_PATH_LEN +
+                              HTTP_CONNECTION_KEEP_ALIVE_VALUE_LEN;
+    setupBuffer( &requestHeaders, expectedHeaders.dataLen );
 
     requestInfo.pPath = 0;
     requestInfo.flags = HTTP_REQUEST_KEEP_ALIVE_FLAG;
-    expectedHeaderLen = HTTP_TEST_PREFIX_HEADER_LEN -
-                        HTTP_TEST_REQUEST_PATH_LEN +
-                        HTTP_EMPTY_PATH_LEN +
-                        HTTP_CONNECTION_KEEP_ALIVE_VALUE_LEN;
-    numBytes = snprintf( expectedHeader, expectedHeaderLen + 1,
+    numBytes = snprintf( ( char * ) expectedHeaders.buffer, sizeof( expectedHeaders.buffer ),
                          HTTP_TEST_HEADER_FORMAT,
                          HTTP_METHOD_GET, HTTP_EMPTY_PATH,
                          HTTP_PROTOCOL_VERSION,
                          HTTP_USER_AGENT_FIELD, HTTP_USER_AGENT_VALUE,
                          HTTP_HOST_FIELD, HTTP_TEST_HOST_VALUE,
                          HTTP_CONNECTION_FIELD, HTTP_CONNECTION_KEEP_ALIVE_VALUE );
-    TEST_ASSERT_EQUAL( numBytes, expectedHeaderLen );
-
-    requestHeaders.pBuffer = httpBuffer;
-    requestHeaders.bufferLen = expectedHeaderLen;
-    test_err = HTTPClient_InitializeRequestHeaders( &requestHeaders, &requestInfo );
-    TEST_ASSERT_EQUAL_MEMORY( requestHeaders.pBuffer, expectedHeader, expectedHeaderLen );
-    TEST_ASSERT_EQUAL( requestHeaders.headersLen, expectedHeaderLen );
-    TEST_ASSERT_EQUAL( test_err, HTTP_SUCCESS );
+    TEST_ASSERT_EQUAL( expectedHeaders.dataLen, numBytes );
+
+    requestHeaders.pBuffer = testBuffer;
+    requestHeaders.bufferLen = expectedHeaders.dataLen;
+    httpStatus = HTTPClient_InitializeRequestHeaders( &requestHeaders, &requestInfo );
+    TEST_ASSERT_EQUAL_MEMORY( requestHeaders.pBuffer, expectedHeaders.buffer,
```

Comment:
The length check should go before the memory check

Reply:
I'll make this change to Http_AddRangeHeader tests as well. Thanks.